build-tools/packages/version-tools/src/bumpTypes.ts

```diff
@@ -86,21 +86,24 @@ export type InterdependencyRange =
  * excluding non-conforming strings than it does at finding valid ones. This appears to be sufficient for our needs,
  * which is good because I don't know how to fix it.
  */
-export function isInterdependencyRange(r: any): r is InterdependencyRange {
-	if (semver.valid(r) !== null) {
+export function isInterdependencyRange(r: unknown): r is InterdependencyRange {
+	if (semver.valid(r as string | semver.SemVer) !== null) {
 		return true;
 	}
 
-	if (RangeOperators.includes(r) || WorkspaceRanges.includes(r)) {
+	if (
+		RangeOperators.includes(r as RangeOperator) ||
+		WorkspaceRanges.includes(r as WorkspaceRange)
+	) {
 		return true;
 	}
 
-	if (semver.validRange(r) === null) {
+	if (semver.validRange(r as string | semver.Range) === null) {
 		return false;
 	}
 
 	if (typeof r === "string") {
-		return RangeOperators.includes(r[0] as any);
+		return RangeOperators.includes(r[0] as RangeOperator);
```

Comment:
Upon re-review I almost like it better as any with a lint disable and comment. `as` is more or less evil and should not be promoted as good generally. This is not a good use of `as`. I recognize `any` as bad and this doesn't really improve type safety.

Reply:
We could also make the arguments strongly typed. I wasn't planning on tackling that for most packages for compatibility reasons, but it might make sense here. Do you know of any reasons that would prevent us from taking a type union instead of `any` or `unknown`?

Reply:
Fixed with type-narrowing

Reply:
I would say that `unknown` doesn't change outside. There is nothing that you could pass before that you can't pass now (per type check) (not even `never`).
The benefit of `unknown` is for the inside to make sure it is doing type checks. In this case every single use had been with a cast; so, the only thing that happened was an appearance of doing the right thing.
The narrowing looks good. The `as` use in the other functions is okay because `includes` is a concrete list and while the warning about "you better use the same type" when call `includes` has merit it isn't necessary in these use cases. It is a good practice to always make sure to isolate such uses in their own helper as done here.

Reply:
Agreed that the user-facing behavior ends up the same for _inputs_. For outputs, the story is very different (irrelevant here).

But as a general rule, I think it's important to avoid allowing `any` to appear in API boundaries (inputs & outputs) both for safety and consistency. Taking unknown instead of any encourages type narrowing (as you suggested). But even if the implementation turns around and casts the unknown to any, that becomes an explicit choice that is easy to argue with in a PR review, etc.

- A long way of me saying that I agree 😋